joblib/numpy_pickle.py

```diff
@@ -229,9 +232,15 @@ def read_mmap(self, unpickler):
             )
             warnings.warn(message)
 
-        return _ensure_native_byte_order(marray)
+        if ensure_native_byte_order:
+            # TODO: should issue a warning here ?
+            # because the array gets copied in the process.
+            # which is precisely what memmap users doesn't want.
+            marray = _ensure_native_byte_order(marray)
```

Comment:
I think there is no point in changing `read_mmap` to ever call `_ensure_native_byte_order`: it's not possible to swap the byte ordering without making a memory copy and the point of loading pickled array with memory mapping is to not allocate new memory but to memory map the buffer.